### ISSUE TITLE
Where I should input CaseId?

### ISSUE THREAD

**Issue by RuslanAisin** [PeterNgTr/codeceptjs-testrail#3]
I Have this error 'Please check {"error":"Field :case_id is not a valid test case."}'
Where in my test I should input value caseId? in conf.js?

**Comment by PeterNgTr** [comment]
Hi, not in config but in scenario like this

Note: you should include the test case id to make it works, otherwise, this plugin has no clue which case id to be added to test run on Testrail.

An example:

Scenario('Search function is displayed @C12345', (I, homePage) => {
  I.seeElement(homePage.searchTextbox);
  I.seeElement(homePage.searchButton);
});

**Comment by RuslanAisin** [comment]
It's succsed. Thank you very usefull!

**Comment by RuslanAisin** [comment]
So I have one more question. Can I select certain cases? At the moment Test run included all cases.

**Comment by PeterNgTr** [comment]
I think that as long as you put the test case ids in your scenarios, the test run will include what you put. :)

**Comment by RuslanAisin** [comment]
In first screen all included cases in my project.
![nobag](https://user-images.githubusercontent.com/49075028/59930232-2bcfaf80-944b-11e9-88f9-f1e36b277691.PNG)
In second screen result of my started test, where I included just three cases. And this cases was tested with result pass or failure. Other cases with status untest I did not include in my *_test.js.
![nobag2](https://user-images.githubusercontent.com/49075028/59930380-781aef80-944b-11e9-99f2-feab21f49e63.PNG)

Thank for help!

**Comment by PeterNgTr** [comment]
Oh I see the problem. It’s by default when adding a run all tcs are included. I would make a fix to publish only executed tcs.

**Comment by RuslanAisin** [comment]
It will be great thank you

**Comment by PeterNgTr** [comment]
@RuslanAisin v1.1.8 should fix this. Can you try and close this issue if it is working for you?

**Comment by RuslanAisin** [comment]
I testes this issue. As you see it a little be worse.
![nobag3](https://user-images.githubusercontent.com/49075028/60020611-0801a900-9699-11e9-81a1-a954aded4189.PNG)
 I run 4 tcs but in TestRail added just one tcs with failure.

**Comment by RuslanAisin** [comment]
I think the problem in 47 Row:
testrail.addRun(projectId, { suite_id: suiteId, name: runName, **include_all: false** }

Maybe you can add logical function that add tcs in TestRail which include *_test.js

**Comment by PeterNgTr** [comment]
@RuslanAisin would be also nice if you can make a pr :)

**Comment by RuslanAisin** [comment]
@PeterNgTr what is 'pr'?

**Comment by PeterNgTr** [comment]
Pull Request:)

**Comment by RuslanAisin** [comment]
@PeterNgTr unfortunetly I dnk, how to do it
![pr](https://user-images.githubusercontent.com/49075028/61124316-3829b400-a4af-11e9-8b6a-022c7d72cf42.png)

**Comment by PeterNgTr** [comment]
@RuslanAisin could you help testing with latest version? I hope this will fix your issue.

**Comment by RuslanAisin** [comment]
Oh hello thank you a lot. Take me timeОтправлено со смартфона Samsung Galaxy.
-------- Исходное сообщение --------От: Peter Nguyen Tr <notifications@github.com> Дата: 04.09.2019  11:05  (GMT+03:00) Кому: PeterNgTr/codeceptjs-testrail <codeceptjs-testrail@noreply.github.com> Копия: RuslanAisin <ruslan-spb@mail.ru>, Mention <mention@noreply.github.com> Тема: Re: [PeterNgTr/codeceptjs-testrail] Where I should input CaseId? (#3) @RuslanAisin could you help testing with latest version? I hope this will fix your issue.

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/PeterNgTr/codeceptjs-testrail/issues/3?email_source=notifications\u0026email_token=ALWNGVGWX7KFVHK3RNVKLM3QH5T4JA5CNFSM4H2P2PUKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD52XJHI#issuecomment-527791261",
"url": "https://github.com/PeterNgTr/codeceptjs-testrail/issues/3?email_source=notifications\u0026email_token=ALWNGVGWX7KFVHK3RNVKLM3QH5T4JA5CNFSM4H2P2PUKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD52XJHI#issuecomment-527791261",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by RuslanAisin** [comment]
@PeterNgTr 
Maybe its only my problem?

'There aren't any tests in this test run.' in TestRail

My Scenario

Feature('Test');

Scenario('test something @C61', (I) => {

    I.amOnPage('/');
});

Scenario('test @C120', (I) => {

    I.amOnPage('/');
});


Scenario('test @C59', (I) => {

    I.amOnPage('/');
});

Scenario('test @C9', (I) => {

    I.amOnPage('/');
});

**Comment by PeterNgTr** [comment]
@RuslanAisin maybe it's due lowercase C? could you try to test with all uppercase C?

**Comment by RuslanAisin** [comment]
@PeterNgTr yeah I tried different cases. Nothing :(

**Comment by PeterNgTr** [comment]
@RuslanAisin are those Ids from testrail or just bogus ids? 🤔

**Comment by RuslanAisin** [comment]
@PeterNgTr , it is real ID from my testrail's cases

**Comment by PeterNgTr** [comment]
@RuslanAisin could you expose your config? (yeah of course you should hide any sensitive info)

**Comment by RuslanAisin** [comment]
@PeterNgTr 
My config
``// ================TestRail====================

  testrail: {
       require: 'codeceptjs-testrail',
       host: 'https://aisin1.testrail.io/',
       user: 'aisin.r@n-market.pro',
       password: 'XKEC88LhJPUp7CL',
      // suiteId: 1,
       projectId: 1,
       runName: '', //input any name
       enabled: true, //false
 },``

Example cases
![image](https://user-images.githubusercontent.com/49075028/64953356-4ae3af00-d88b-11e9-9206-c2c57e7c3e2e.png)

Example test

``
Feature('Federal Page');

Scenario('Central button @C58', (I) => {
	I.amOnPage('/');
	I.seeInTitle('***');
	I.see('Нмаркет.ПРО, 2019');
	I.click('//*[@class="main-center-block__title"]'); //ценральная плашка
	I.wait(1);
	I.seeElement('//*[@class="b-big-button b-big-button_red"]');
});

Scenario('Absent City @C111', (I) => {
	I.amOnPage('/');
	I.click('//*[@class="b-big-button b-big-button_red"]'); // Моего города нет
	I.wait(1);
	I.seeInTitle('***');

});

Scenario('Link Industry @C112', (I) => {
	I.amOnPage('/');
	I.click('//*[@class="g-made-by"]'); // Индустрия Медиа - it технологии для бизнеса
	I.wait(1);
	I.switchToNextTab();
    I.seeInTitle('***');
	I.closeCurrentTab();
	I.click('//*[@class="b-link"]'); 
	I.wait(1);
	I.switchToNextTab();
	I.seeInTitle('***');
	I.closeCurrentTab();

});

Scenario('AppStore @C63', (I) => {	
	I.amOnPage('/');
	I.click('//div[@class="b-footer-mobile-btns"]/a[contains(@href, "apple.com")]'); // AppStore
	I.wait(1);
	I.switchToNextTab();
	I.see('***');
	I.closeCurrentTab();

});

Scenario('GooglePlay @C64', (I) => {	
	I.amOnPage('/');
	I.click('//div[@class="b-footer-mobile-btns"]/a[contains(@href, "google.com")]'); // GoogleMarket
	I.wait(1);
	I.switchToNextTab();
	I.see('***');
	I.closeCurrentTab();
});
	
Scenario('SupportForm @C65', (I) => {
	I.amOnPage('/');
	I.seeInTitle('***');
	I.see('***');
	I.click('//*[@class="ask-support b-feedback-trigger b-link"]'); //Сообщить об ошибке в службу поддержки 
	I.wait(1);
	I.see('***');
	I.fillField('//*[@id="userEmail"]', 'Test@test.com');
	I.fillField('//*[@id="phone"]', '77777777777');
	I.fillField('//*[@id="messageText"]', 'Automat test system. Do not answer this message.');
	I.click('//*[@class="b-orange-button b-orange-button_with-one-decor sendfeedBackForm"]');
	I.wait(1);
	I.see('Сообщение отправлено');
	I.click('//*[@class="popup-close"]');
});	

Scenario('All Regions @C110', (I) => {
	I.amOnPage('/');
	I.seeInTitle('***');
	I.see('***');
	I.openNewTab();
	I.amOnPage('/');
	I.wait(3);
	I.seeInTitle('***');
	I.closeCurrentTab();
});

Scenario('About Project @C59', (I) => {
	I.amOnPage('/');
	I.click('//*[@class="b-section-property__btn b-big-button"]'); //О проекте
	I.wait(3);
	I.seeInTitle('***');
	
});

Scenario('How to Search @C69', (I) => {	
	I.amOnPage('/');
	I.click('//*[@onclick="window.regionSwitch.changeRegion(78)"]');
	I.wait(3);
	I.seeInTitle('***');
	I.click('//*[@class="b-search-help"]'); //Как искать
	I.wait(1);
	I.see('***');
	I.seeNumberOfElements('//*[contains(@class, "search-popup-dialog")]/*[contains(@class, "ui-dialog-titlebar")]/button', 1);
	I.click('//*[contains(@class, "search-popup-dialog")]/*[contains(@class, "search-popup-wrapper")]/a[contains(@class, "b-orange-button_small_close")]');  // Закрытие формы
});

``

**Comment by PeterNgTr** [comment]
@RuslanAisin could you run with --verbose mode then adding the logs here?

**Comment by RuslanAisin** [comment]
@PeterNgTr  I attached file

[new 1.txt](https://github.com/PeterNgTr/codeceptjs-testrail/files/3616069/new.1.txt)

**Comment by PeterNgTr** [comment]
@RuslanAisin are you updating your package.json to use latest version of this helper?

**Comment by RuslanAisin** [comment]
@PeterNgTr  yeah "version": "1.2.1",

**Comment by RuslanAisin** [comment]
@PeterNgTr NodeJS v8.10.0 mb this problem?

**Comment by PeterNgTr** [comment]
I didn't see anything wrong in your log file. Would be nice if you could expose your demo project, I would run on my local to debug it.

**Comment by RuslanAisin** [comment]
@PeterNgTr  so am I, how do you thing about node version?

**Comment by PeterNgTr** [comment]
Not really sure about it. But let’s try. :)

**Comment by RuslanAisin** [comment]
@PeterNgTr  your test case works properly and correct adding to runs?

**Comment by PeterNgTr** [comment]
@RuslanAisin yeah, it worked as expected.

**Comment by PeterNgTr** [comment]
also being tested here. https://github.com/PeterNgTr/codeceptjs-testrail/issues/1#issuecomment-530250018

**Comment by RuslanAisin** [comment]
@PeterNgTr what is your version node?

**Comment by PeterNgTr** [comment]
@RuslanAisin it's node:12.4.0

**Comment by RuslanAisin** [comment]
> I didn't see anything wrong in your log file. Would be nice if you could expose your demo project, I would run on my local to debug it.

What should I share? my testrail account or test?

**Comment by RuslanAisin** [comment]
@PeterNgTr I've checked your node version, it doesnt help

**Comment by PeterNgTr** [comment]
> > I didn't see anything wrong in your log file. Would be nice if you could expose your demo project, I would run on my local to debug it.
> 
> What should I share? my testrail account or test?

@RuslanAisin it's better that you can provide me an example code on github. Or, you can contact me with skype id: peter.nguyentr@gmail.com for fast communication.

**Comment by RuslanAisin** [comment]
@PeterNgTr oh my God it is works!!!! Its very very strange i dont know maybe miracle or smth like that))))))
![image](https://user-images.githubusercontent.com/49075028/64962440-1b8b6d00-d8a0-11e9-886d-5696494a1328.png)

**Comment by RuslanAisin** [comment]
@PeterNgTr I have one thing. I deleted all milestones. Maybe it was problem. Need more test. In this time all works property. Thank you for help. I close the issue

**Comment by PeterNgTr** [comment]
@RuslanAisin cool. 👍

**Comment by yayura** [comment]
Hi!
Can you help, please? 
 When I use [tags](https://codecept.io/advanced/#tags) in my tests, like this
`Scenario('test1 @C1', (I) => {
I.amOnPage('/');
}).tag('@test').tag('@inprogress');`
plugin does not submits results to TestRail,  with error at the end 
`(node:8484) UnhandledPromiseRejectionWarning: Error: Request failed with status code 400
    at createError (C:\Users\User\Documents\codeceptJS\node_modules\axios\lib\core\createError.js:16:15)
    at settle (C:\Users\User\Documents\codeceptJS\node_modules\axios\lib\core\settle.js:17:12)
    at IncomingMessage.handleStreamEnd (C:\Users\User\Documents\codeceptJS\node_modules\axios\lib\adapters\http.js:236:11)
    at IncomingMessage.emit (events.js:215:7)
    at IncomingMessage.EventEmitter.emit (domain.js:476:20)
    at endReadableNT (_stream_readable.js:1183:12)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
(node:8484) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:8484) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit`

**Comment by PeterNgTr** [comment]
Hey, do you check if you can put .tag.tag like that with codeceptjs?

**Comment by yayura** [comment]
@PeterNgTr  We use tags like this: 
`Feature ('Login')`
`Scenario('test1 @C1', (I) => {`
`I.amOnPage('/');`
`tag('@inprogress').tag('@test').tag('@stage');`

I tried to add  .tag('@C1') to the end and to the start of tag chain,
`tag('@C1').tag('@test').tag('@stage');`
`tag('@inprogress').tag('@test').tag('@C1');`

Actual result: 
Testcases were not added to the testrun and results was not submitted, 
Error as in my  previous comment
` UnhandledPromiseRejectionWarning: Error: Request failed with status code 400`

If I removed tags from first scenario and left in next scenarios - same result, nothing added. 
If I removed all tags from file - ok! tests added to the testrun and result submitted correctly 

Thanks!
